### PR TITLE
Show dialog when logging out and suggestive text in product detail

### LINKED ISSUE #196: Should ask for confirmation before logging out
When tapped on logout button, should ask for confirmation.

### LINKED ISSUE #213: Main menu misaligned with XD
Fix the following where the current main menu is different from the XD design:
1. Product list - logo and text
2. Logout button - logo

Current customer main menu:

Screen Shot 2020-04-08 at 20 33 16

Current admin main menu:

Screen Shot 2020-04-08 at 21 53 59

### LINKED ISSUE #217: Add suggestive text in customer product detail page
Screen Shot 2020-04-08 at 20 56 51

### PR DESCRIPTION
**Pull request by dragonfly91**
Demo: https://youtu.be/t7rqwRWjr4Q

Fixes #196 
Fixes #213
Fixes #217  

Note: Unable to change the logout icon as this was not available. 

Screen Shot 2020-04-10 at 22 01 16
Screen Shot 2020-04-11 at 19 06 02
Screen Shot 2020-04-11 at 19 03 47
Screen Shot 2020-04-11 at 19 06 34
Screen Shot 2020-04-11 at 19 06 21

### CODE REVIEW

File: lib/screens/customer_product_detail_page.dart [hunk 1]
```diff
@@ -84,18 +84,44 @@ class _CustomerProductDetailPageState extends State<CustomerProductDetailPage> {
...
 
+  String _getSelectStockDisplayText(StockType productStockType) {
+    if (productStockType == StockType.colorOnly)
+      return "下記より色をお選びください";
+    else if (productStockType == StockType.sizeAndColor)
+      return "下記よりサイズと色をお選びください";
+    else if (productStockType == StockType.sizeOnly)
+      return "下記よりサイズをお選びください";
+    else
+      return null;
```
Comment (kumilange) [inline lib/screens/customer_product_detail_page.dart:116]: @dragonfly91 How do you differentiate `switch` and `if/else`? 🙂

Comment (dragonfly91) [reply lib/screens/customer_product_detail_page.dart:116]: Hi @kumiko-haraguchi I'm not sure if I understand your question correctly. Do you mean, when to use switch and when to use if/else? If this is your question, I think both are interchangeable, but I felt the if-else implementation to be more compact in this instance.

Comment (kumilange) [reply lib/screens/customer_product_detail_page.dart:116]: @dragonfly91 I've seen you using `switch` so far and thought you have a reason to chose it. I didn't see any significant difference in this `_getSelectStockDisplayText` and other `get` methods with switch statement so I wondered how you decide when to use `switch` and `if/else`. That'll help me to follow when I encounter the same situation in the future. :)

I didn't get this. Could you elaborate on it?
>the if-else implementation to be more compact in this instance

Comment (dragonfly91) [reply lib/screens/customer_product_detail_page.dart:116]: I think we currently don't have any strong opinion towards using switch vs if/else, so feel free to use whichever makes the most sense to you! In this case, I meant 'compact' in the sense of lines of code :)

Comment (kumilange) [reply lib/screens/customer_product_detail_page.dart:116]: @dragonfly91 I see, ok, got it!!! :D

Comment (dragonfly91) [reply lib/screens/customer_product_detail_page.dart:116]: Awesome! If you're satisfied with this PR, can you please resolve this comment and approve the PR? @kumiko-haraguchi

**Review by kumilange — APPROVED** [review]

**Review by kumilange — APPROVED** [review]

**Pull request merged**